### DISCUSSION TITLE
Error while building Writefreely

### DISCUSSION THREAD

**Discussion by itbeard** [writefreely/writefreely#727] [Questions / Support] [answered]
Hello!
Command `go build -v -tags='sqlite' ./cmd/writefreely/` is throwing error while I am trying to build writefreely:

``` go
# github.com/writefreely/writefreely
./bindata.go:81:6: schemaSql redeclared in this block
	./app.go:928:5: other declaration of schemaSql
./bindata.go:101:6: sqliteSql redeclared in this block
	./app.go:931:5: other declaration of sqliteSql
./bindata.go:203:16: cannot use schemaSql (variable of type string) as func() (*asset, error) value in map literal
./bindata.go:204:16: cannot use sqliteSql (variable of type string) as func() (*asset, error) value in map literal
./bindata.go:253:17: cannot use schemaSql (variable of type string) as func() (*asset, error) value in struct literal
./bindata.go:254:17: cannot use sqliteSql (variable of type string) as func() (*asset, error) value in struct literal
```

I'm using go 1.20 on MacOS environment. Any ideas? I'm like a novice in Go btw, so sorry if this is dumb question for pro)

**Comment by itbeard** [comment, marked as answer]
I found a solution. Just change function names in `bindata.go`
`func schemaSql()` -> `func schemaSql1()`
`func sqliteSql()` -> `func sqliteSql1()`

And change all of calling those functions inside `bindata.go`

I also found several mistakes in "Development setup" article here (maybe it is do not mistakes because I used macos): https://writefreely.org/docs/latest/developer/setup
Do I need to post info about it here?

**Reply by thebaer** [reply]
For any documentation mistakes, please open an issue here! https://github.com/writefreely/documentation

That's where everything is generated from. Appreciate it!

**Comment by glyn** [comment]
@itbeard You shouldn't have to edit the code to get it to build. Did you manage to build it in the end without those edits?

Also, where is bindata.go??

**Reply by itbeard** [reply]
Hello,
`Did you manage to build it in the end without those edits?` - no I do not.
`Also, where is bindata.go??` this file generating while compilation of sources

**Reply by glyn** [reply]
FYI I built the executable for Linux only by hacking the makefile, thus:
```
diff --git a/Makefile b/Makefile
index b230cb7..8b2b7e2 100644
--- a/Makefile
+++ b/Makefile
@@ -24,10 +24,11 @@ build-no-sqlite: deps-no-sqlite
 	cd cmd/writefreely; $(GOBUILD) -v -tags='netgo' -o $(BINARY_NAME)
 
 build-linux: deps
-	@hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
-		$(GOCMD) install src.techknowlogick.com/xgo@latest; \
-	fi
-	xgo --targets=linux/amd64, -dest build/ $(LDFLAGS) -tags='netgo sqlite' -go go-1.19.x -out writefreely -pkg ./cmd/writefreely .
+	# @hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+	#	$(GOCMD) install src.techknowlogick.com/xgo@latest; \
+	# fi
+	$(GOINSTALL) $(LDFLAGS) -tags='netgo sqlite' ./cmd/writefreely
+
 
 build-windows: deps
 	@hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
```
But I haven't managed to build the complete install package this way (e.g. including `/templates`).
